tests/integration/test_leases.py

```diff
@@ -132,7 +132,7 @@ def setup(self):
     def test_valid_lease_id(self, auth_headers):
```

Comment:
These are excellent changes, However, the name of this test is a bit confusing now. We're testing something that should be valid, but asserting a 400 error.

I agree with the change in the logic to say that an empty payload should be a bad request, and since new logic was added we should update our tests accordingly. I'm fine to do that in this ticket or we can create a new ticket for it. What would you like to do?

Reply:
I think test is not necessary. test_date_is_updated_with_valid_payload tests both that the lease id is valid and that an update is made with a valid payload. And test_updatedData_is_left_unchanged_when_given_invalid_parameters tests for a valid lease id with an empty payload, which is essentially the same test as test_valid_lease_id. 

test_invalid_lease_id tests the case where the lease id is not valid. 

I recommend removing test_valid_lease_id.

Reply:
If you agree, I'll just make the change.

Reply:
Or we can rename the test. test_valid_id_empty_payload.

Reply:
Yes, I agree. Let's remove the test.

Reply:
or perhaps we can rename it to `test_valid_id_with_invalid_payload` - I don't think we have any tests that check what happens when there is an invalid payload.

Reply:
nevermind. I need to learn how to read better. 
> And test_updatedData_is_left_unchanged_when_given_invalid_parameters tests for a valid lease id with an empty

Yes, agreed. Let's remove it.